securedrop_client/logic.py

```diff
@@ -661,8 +661,40 @@ def export_file_to_usb_drive(self, file_uuid: str, passphrase: str) -> None:
         if not self.qubes:
             return
 
+        fn_no_ext, dummy = os.path.splitext(os.path.splitext(file.filename)[0])
+        filepath = os.path.join(self.data_dir, fn_no_ext)
         self.export.begin_usb_export.emit([path_to_file_with_original_name], passphrase)
 
+    def print_file(self, file_uuid: str) -> None:
+        '''
+        Send the file specified by file_uuid to the Export VM.
+
+        Once a file is downloaded, it exists in the data directory with the same filename as the
+        server, except with the .gz.gpg stripped off.
+
+        If the file is missing, update the db so that is_downloaded is set to False.
+        '''
+        file = self.get_file(file_uuid)
+        logger.info('Printing file {}'.format(file.original_filename))
+
+        if not self.downloaded_file_exists(file.uuid):
+            self.sync_api()
+            return
+
+        path_to_file_with_original_name = os.path.join(self.data_dir, file.original_filename)
+
+        if not os.path.exists(path_to_file_with_original_name):
+            fn_no_ext, dummy = os.path.splitext(os.path.splitext(file.filename)[0])
+            filepath = os.path.join(self.data_dir, fn_no_ext)
+            os.link(filepath, path_to_file_with_original_name)
+
+        if not self.qubes:
+            return
+
+        fn_no_ext, dummy = os.path.splitext(os.path.splitext(file.filename)[0])
```

Comment:
does this meet the threshold for a helper method since this is repeated 3 times in this snippet?

Reply:
i decided to keep this improvement out of the scope of this pr since the plan is to abstract this out and i believe @redshiftzero is on the case: https://github.com/freedomofpress/securedrop-client/issues/579

Reply:
if you do another round of testing on this PR then I'd put this in a helper method prior to doing that (else we can do in the future) - just because this is a fair bit of duplicated code that it would be nice to DRY while it's fresh in our minds even if we intend to modify it in the future